docs/upgrade/2.0-3.0.md

```diff
@@ -0,0 +1,302 @@
+
+```
+$ mysql -u username -p password database_name < madison_db_dump.sql
+```
+
+### Pull down Madison 3.0 code from GitHub!
+
+You will want to pull in the changes for Madison 3.0 from our main GitHub repo,
+located at https://github.com/opengovfoundation/madison. If your project is not
+set up to track from the primary repo, you will want to add it as a remote:
+
+```
+$ git remote add opengovfoundation git@github.com:opengovfoundation/madison.git
+```
+
+Now you should be able to do:
+
+```
+$ git pull opengovfoundation master
+```
+
+This will pull down the master branch to your local instance. If you're doing
+this in production *please make sure you have your codebase backed up as
+mentioned above*!
+
+This should put the new folder structure in place with your local project. To
+confirm, make sure the top level directory has `client` and `server` folders,
+and **does not** have a top level  `public` directory.
+
+### Configuration
+
+Previously, the primary configuration file is `.env` in the top level directory.
+That file now resides in the `server` folder. You will want to move it to be in
+the `server` folder and also check `server/.env.example` to make sure all
+necessary things are configured.
+
+### Apache configuration
+
+You will need to update the apache configuration to match the new folder
+structure. For a reference of what this should look like, there is an example
+configuration file [included in the docs](/docs/apache.conf.example).
+
+Again, we recommend that you back up a copy of your current configuration.
+
+### Run database migrations
+
+Make sure to run database migrations, as there have been many changes in 3.0.
+Some of the notable changes are:
+
+* A new `jobs` table has been added for queue working
+* Independent Sponsorship has been moved to a group only model
+* New table for pages and page content
+* Unification of data model for annotations / comments / replies
+
+For your convenience, a top level Makefile contains the command necessary to
+migrate your database:
+
+```
+$ make db-migrate
+```
+
+Alternatively, if you deploy using Capistrano (recommended) it should
+automatically run the migrations after a successful deployment.
+
+
+### Queue worker
+
+In Madison 3.0, work has been done to make notifications functional. These
+notifications are run in background jobs and require **queue workers** to be
+running on the server for them to be processed.
+
+There are a number of ways to go about this, but our recommendation is to use
+[supervisor](http://supervisord.org/introduction.html) to ensure that the queue
+workers are always running on your system.
+
+You should use your OS's package manager to install supervisor:
+
+```
+$ sudo apt-get install supervisor
+```
+
+From there, an example supervisor configuration looks like so:
+
+```
+[program:laravel-worker]
+process_name=%(program_name)s_%(process_num)02d
+command=php /path/to/madison/server/artisan queue:work sqs --sleep=3 --tries=3
```

Comment:
I think we're fine to use the listeners instead then. Thanks for clarifying here! Should have checked in with you first since you've done this work.